cmd/identity/keys.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/spf13/cobra"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/storj"
+)
+
+var (
+	keysCmd = &cobra.Command{
+		Use:   "keys",
+		Short: "Manage keys",
+	}
+	keyGenerateCmd = &cobra.Command{
+		Use:   "generate",
+		Short: "generate lots of keys",
+		RunE:  cmdKeyGenerate,
+	}
+
+	keyCfg struct {
+		MinDifficulty int    `help:"minimum difficulty to output" default:"18"`
+		Concurrency   int    `help:"worker concurrency" default:"4"`
+		OutputDir     string `help:"output directory to place keys" default:"."`
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(keysCmd)
+	keysCmd.AddCommand(keyGenerateCmd)
+	cfgstruct.Bind(keyGenerateCmd.Flags(), &keyCfg)
+}
+
+func cmdKeyGenerate(cmd *cobra.Command, args []string) (err error) {
+	ctx := process.Ctx(cmd)
+	err = os.MkdirAll(keyCfg.OutputDir, 0755)
+	if err != nil {
+		return err
+	}
+	counter := new(uint32)
```

Comment:
purely out of curiosity, why `counter := new(uint32)` vs. `var counter int32 = 0` ?

Reply:
i'm probably unreasonably worried about memory alignment issues:

https://golang.org/pkg/sync/atomic/#pkg-note-BUG says:
> On both ARM and x86-32, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.

this isn't 64 bit, so i don't really have a good reason here, but in general when using sync/atomic i've just been making it a habit